### PR TITLE
Scala Native support

### PR DESCRIPTION
**Pull request by ysthakur**
This branch is for eventually creating Vyxal executables directly using Scala Native. There's also GraalVM, which we should check out too. Executables made by Scala Native will likely be smaller, since they don't contain all the Java stuff that GraalVM has. Scala Native will also be able to optimize more, since it's aware of the original Scala code, while GraalVM works on bytecode. However, the size and performance difference probably won't be big enough to be noticeable, and GraalVM has been around longer than Scala Native and works on JARs, so we may just want to go with that.

### CODE REVIEW

File: build.sc [hunk 1]
```diff
@@ -47,7 +49,7 @@ trait VyxalModule extends SbtModule {
 }
 
 /** Shared and JVM-specific code */
-object jvm extends VyxalModule {
+object jvm extends ScalaModule with VyxalModule {
```
Comment (ysthakur) [inline build.sc:52]:
```suggestion
object jvm extends VyxalModule {
```

**Review by Lyxal — APPROVED** [review]
I think the fact that everything has to be JS compatible means that code we write will fit just fine into the requirements of Native. As in, we won't be using anything Java anyway.

**Comment by Lyxal** [comment]
besides, we can always figure out any issues when we come across them + decide to not support native if need be

**Comment by Lyxal** [comment]
merge when you're ready

**Pull request merged**